smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/utils/AuthUtils.kt

```diff
@@ -0,0 +1,39 @@
+package software.amazon.smithy.swift.codegen.utils
+
+import software.amazon.smithy.model.knowledge.ServiceIndex
+import software.amazon.smithy.model.traits.HttpBearerAuthTrait
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.config.DefaultProvider
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.swiftmodules.SmithyHTTPAuthTypes
+
+open class AuthUtils(
+    private val ctx: ProtocolGenerator.GenerationContext
+) {
+    val authSchemesDefaultProvider = DefaultProvider(
+        { getModeledAuthSchemesSupportedBySDK(ctx, it) },
+        isThrowable = false,
+        isAsync = false
+    )
+
+    fun getModeledAuthSchemesSupportedBySDK(
+        ctx: ProtocolGenerator.GenerationContext,
+        writer: SwiftWriter,
+    ): String {
+        val effectiveAuthSchemes = ServiceIndex(ctx.model).getEffectiveAuthSchemes(ctx.service)
+        var authSchemeList = arrayOf<String>()
+
+        if (effectiveAuthSchemes.contains(HttpBearerAuthTrait.ID)) {
+            authSchemeList += writer.format("\$N()", SmithyHTTPAuthTypes.BearerTokenAuthScheme)
+        }
+
+        authSchemeList = addAdditionalSchemes(writer, authSchemeList)
+
+        return "[${authSchemeList.joinToString(", ")}]"
```

Comment:
consider changing to `return addAdditionalSchemes(writer, authSchemeList).joinToString(prefix = "[", postfix = "]")` or `return authSchemeList.joinToString(prefix = "[", postfix = "]")` to use idiomatic kotlin-provided functions -- prefix / postfix makes it more clear what's being added

the `", "` in joinToString() should be unnecessary as comma separation is the default behavior (https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/join-to-string.html)

Reply:
Changed as suggested!